.github/workflows/release_github.yml

```diff
@@ -39,119 +85,120 @@ jobs:
-          echo "Cannot release from main branch, please select valid release branch."
-          exit 1
-
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          token: ${{ secrets.GH_PAT }}
           ref: main
           fetch-depth: 0
-
       - name: Configure git
         run: |
-          git config user.email "actions@github.com"
+          git config user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
           git config user.name "GitHub Actions"
-          git pull
+          git config -l
 
-      - name: Merge changes into main
+      - name: Merge branch into main
         run: |
           git switch main
-          git branch ${{ github.ref_name }} origin/${{ github.ref_name }} 
-          git merge ${{ github.ref_name }} --no-ff --no-commit
-          git commit --no-edit
-
-      - name: Bump version
+          git branch -f ${{ github.ref_name }} origin/${{ github.ref_name }} 
+          git merge ${{ github.ref_name }} --no-ff --no-edit
+    
+      - name: Install Python
+        uses: actions/setup-python@v5.1.1
+        with:
+          python-version: '3.12'
+  
+      - name: Install bump-my-version
+        shell: bash
+        run: pip install "bump-my-version==0.28.1"
+        
+      - name: Pass Inputs to Shell
         id: bump
+        shell: bash
         run: |
-          echo "-- install bump-my-version"
-          python3 -m pip install bump-my-version
-          echo "-- bump the version"
-          bump-my-version bump ${{ github.event.inputs.version_level }} --commit --tag
-          echo "-- push bumped version"
-          echo "RELEASE_TAG=$(git describe --tags --abbrev=0)" >> $GITHUB_OUTPUT
-          git push --tags -f
-          git push
+          echo "previous-version=$(bump-my-version show current_version)" >> $GITHUB_OUTPUT
+          bump-my-version bump ${{ inputs.version_level }} --commit --tag
+          ([[ $? -gt 0 ]] && echo "bumped=false" || echo "bumped=true") >> $GITHUB_OUTPUT
+          echo "current-version=$(bump-my-version show current_version)" >> $GITHUB_OUTPUT
+      - name: Fail if bumping failes
+        if: steps.bump.outputs.bumped == 'false'
+        run: |
+          echo "Bumping failed."
+          git reset --hard HEAD^
+          exit 1
+      - name: Check new version number
+        if: steps.bump.outputs.bumped == 'true'
+        run: |
+          echo "Version was bumped from ${{ steps.bump.outputs.previous-version }} to ${{ steps.bump.outputs.current-version }}!"
 
+      - name: Merge main into develop
+        run: |
+          git checkout develop
+          git merge main --no-ff --no-edit
+      
+      - name: Push changes to GitHub
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: develop
+
+      - name: Checkout main
+        run: |
+          git checkout main
+
+      - name: Push changes to GitHub
```

Comment:
```suggestion
      - name: Push changes to main
```

Reply:
so, at this point, if merging to main fails, then dev has pushed changes (at the very least the version bump), right?

Reply:
No. The merging has already taken place (locally). This only pushes the changes to GitHub. 

The only way this would is if between the checkout and this step (which is less than a second) `main` would change.